[Docs] Update dependencies for docs

Michaelvll opened the pull request:
<!-- Describe the changes in this PR -->

This is following the docs dependency update by https://github.com/ray-project/ray/releases/tag/ray-2.9.0

This PR also fixes the dark mode.

<!-- Describe the tests ran -->
<!-- Unit tests (tests/test_*.py) are part of GitHub CI; below are tests that launch on the cloud. -->

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
  - Render the docs locally
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

concretevitamin reviewed (commented):
Thanks, any major differences in the rendered HTML?

Michaelvll commented:
> Thanks, any major differences in the rendered HTML?

The new depdencies slightly improves the looking of the rendered HTML, and adds the dark mode.
This PR:
<img width="1240" alt="image" src="https://github.com/skypilot-org/skypilot/assets/6753189/fd7cf722-d589-4339-bb0d-a068842d72e6">
<img width="1296" alt="image" src="https://github.com/skypilot-org/skypilot/assets/6753189/ca4edb74-66a4-492e-8120-ed6a0e6b7c79">

Original:
<img width="1319" alt="image" src="https://github.com/skypilot-org/skypilot/assets/6753189/3ae912ba-7c96-4f05-9e03-1f812bbeb7e8">

Michaelvll commented on docs/requirements-docs.txt:
```diff
@@ -1,16 +1,18 @@
+myst-parser==2.0.0
+
+sphinx-autodoc-typehints==1.25.2
+sphinx-book-theme==1.1.0
```
I suppose it should be fine for supporting less python version for the docs, as it will only be built by our hosted docs?

concretevitamin replied:
It'd make devs harder since we need to create a new env for rendering htmls locally. Do we definitely require sphinx-book-theme==1.1.0?

concretevitamin replied:
Oops, GitHub only now sends out the email notif. Sg.

romilbhardwaj commented:
Thanks @Michaelvll!
I switched to a new conda env for building and it worked fine. 

Things I like:
* Like the native dark mode support
* Looks more modern
* The font is slightly smaller so more of the TOC fits in the screen
* Small indent in TOC makes it nicer and cleanly separated

Minor feedback (not necessary to fix, but nice to have):
### Colors

1. The mouseover color is a bright orange and underlines the text, seems a little jarring
<img width="368" alt="image" src="https://github.com/skypilot-org/skypilot/assets/4416605/02c9804f-a935-4d5d-860a-dcbaf6129e68">


2. The code block font colors are harder to read than our current codeblock. See image below, left is new, right is old.
<img width="1511" alt="image" src="https://github.com/skypilot-org/skypilot/assets/4416605/0abd43ea-f893-41dd-a213-d3f8a159b6f2">

Is it possible to use the older color scheme?

### TOC at the bottom of the landing page
Refer to image below. Left is new, right is old. The new one seems a little dense - any way to add more space there?
<img width="1509" alt="image" src="https://github.com/skypilot-org/skypilot/assets/4416605/fc2c4962-d9ed-45e1-9aad-48420f856b77">

Michaelvll commented:
Hey @romilbhardwaj, thank you for the suggestions! I have reverted the code style to the original theme, and changed the hover color to the same color as the original one. For the TOC, I could not find a easy way to increase the space. Should we merge this PR first and then figure out a way to fix that? : )

concretevitamin commented:
This new version looks great!

A few nits to see if there are knobs to get rid of them:

1. Purple border around search bar: this is automatically shown when I click on the top-left search bar.
<img width="835" alt="Screen Shot 2024-01-15 at 11 28 21" src="https://github.com/skypilot-org/skypilot/assets/592670/31ae5e99-5c0c-4bf8-8241-e9a025e24d7e">

2. The left-side nav bar automatically has the scroll bar displayed even when the mouse is not in the nav bar area, see pic:
<img width="300" alt="Screen Shot 2024-01-15 at 11 29 16" src="https://github.com/skypilot-org/skypilot/assets/592670/82c03a79-78b6-4565-acd7-eabfa6fb6cb4">

This is not the case for the current UI.

3. On https://skypilot.readthedocs.io/en/update-docs-dependency/, bottom of nav bar, there's an ad displayed by RTD. Is that expected? Maybe it's for preview sites? Our live docs site doesn't have the ad.

Michaelvll commented:
> This new version looks great!
> 
> A few nits to see if there are knobs to get rid of them:
> 
> 1. Purple border around search bar: this is automatically shown when I click on the top-left search bar.
> 
> <img alt="Screen Shot 2024-01-15 at 11 28 21" width="835" src="https://private-user-images.githubusercontent.com/592670/296854901-31ae5e99-5c0c-4bf8-8241-e9a025e24d7e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU4NjIyNTYsIm5iZiI6MTcwNTg2MTk1NiwicGF0aCI6Ii81OTI2NzAvMjk2ODU0OTAxLTMxYWU1ZTk5LTVjMGMtNGJmOC04MjQxLWU5YTAyNWUyNGQ3ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyMVQxODMyMzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NzNiMjMyMTJjMjNkNWNkMWEzZGRiNjk1NDFiODA4NzE2MmMyMWI1OWJkNjRjNmY2OGFlYmMxNjc4OWQwMzEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.9JAc35blUKz5gOMENWZR81U6GyZVFcXS-DGgE7y2GOk">
> 2. The left-side nav bar automatically has the scroll bar displayed even when the mouse is not in the nav bar area, see pic:
> 
> <img alt="Screen Shot 2024-01-15 at 11 29 16" width="300" src="https://private-user-images.githubusercontent.com/592670/296855038-82c03a79-78b6-4565-acd7-eabfa6fb6cb4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU4NjIyNTYsIm5iZiI6MTcwNTg2MTk1NiwicGF0aCI6Ii81OTI2NzAvMjk2ODU1MDM4LTgyYzAzYTc5LTc4YjYtNDU2NS1hY2Q3LWVhYmZhNmZiNmNiNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyMVQxODMyMzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NDA4YjNjNTE0NDkyZmJmNDRjMDlhNjA5OTA2NzY5OTRkYmNhYzAxMmNmMGVhMWZhMzNkOWMwZjZmMDA3YTk4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.os7ZN7OnRqihtTgpm7DIayT0onoS8dylx2w6KgGOIOg">
> This is not the case for the current UI.
> 
> 3. On https://skypilot.readthedocs.io/en/update-docs-dependency/, bottom of nav bar, there's an ad displayed by RTD. Is that expected? Maybe it's for preview sites? Our live docs site doesn't have the ad.

Thanks for the suggestions! I fixed all of them. PTAL.

concretevitamin reviewed (approved):
LGTM, thanks @Michaelvll!
